### ISSUE TITLE
Support cargo-bpf

### ISSUE THREAD

**Issue by naomijub** [xd009642/tarpaulin#763]
**Describe the feature**

We have been working with Solana library and we have a significant amount of our tests been executed with the command `cargo test-bpf`, would it be possible to determine que test command we want to execute for the test coverage?

So instead of executing `cargo test` we can pass the command `cargo test-bpf`.

Links:
[cargo-bpf](https://ingraind.org/api/cargo_bpf/)

**Comment by naomijub** [comment]
I would suggest a custom field in `--run-types` that support a string to be executed:

```
--run-types <TYPE>...        Type of the coverage run [possible values: Tests, Doctests, Benchmarks, Examples,
                                     Lib, Bins, AllTargets, Custom(String)]
```

**Comment by xd009642** [comment]
yeah it should be do-able, I am curious how well (if) ptrace will work with bpf but if it does I'm happy to add this :+1:

**Comment by naomijub** [comment]
Hope it works, but being able to pass test command may help future developers as well

**Comment by SteMak** [comment]
> I would suggest a custom field in `--run-types` that support a string to be executed:

I like the proposal, would it be implemented in the near future?

An ability to check the coverage of any `cargo run/test-bpf/custom-executor` call sounds impressive!

**Comment by xd009642** [comment]
I've started adding custom runners in https://github.com/xd009642/tarpaulin/pull/1866 this is a necessary part of bpf support you can specify a config with a test launcher that runs it. After that I'll look at the next steps (this feature also unblocks wasm support)

**Comment by xd009642** [comment]
Looking into it a bit more I don't think the LLVM coverage would work as-is in bpf. But if you can compile a program with something like https://github.com/Amanieu/minicov maybe with a custom test runner you could print out the coverage information and read it from the bpf test process and then serialize it to a file and it would work with tarpaulin using the custom runner support.

Leaving this as a note for anyone who wants to try it out (or future me)

**Comment by naomijub** [comment]
You may close the issue, I'm not working on this anymore and you already gave a solution. So, all good on my end
